File: Notes/app/src/main/java/com/serhii/apps/notes/control/managers/AppForegroundListener.java
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */

package com.serhii.apps.notes.control.managers;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;

import com.serhii.core.log.Log;

/**
 * Listener for foreground / background transitions.
 */
public class AppForegroundListener implements LifecycleObserver {

    private static final String TAG = "AppForegroundListener";

    private static boolean isInForeground;

    public static boolean isInForeground() {
        return isInForeground;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    public void onForegroundEnter() {
        Log.info(TAG, "onForegroundEnter()");
        isInForeground = true;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    public void onForegroundExit() {
        Log.info(TAG, "onForegroundExit()");
        isInForeground = false;
    }
}

File: Notes/app/src/main/java/com/serhii/apps/notes/control/AppForegroundListener.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.apps.notes.control

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import com.serhii.core.log.Log.Companion.info
import java.util.concurrent.atomic.AtomicBoolean

/**
 * Listener for foreground / background transitions.
 */
object AppForegroundListener : LifecycleObserver {

    private const val TAG = "AppForegroundListener"
    private val isInForeground = AtomicBoolean(false)

    fun isInForeground() = isInForeground.get()

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onForegroundEntered() {
        info(TAG, "onForegroundEntered()")
        isInForeground.set(true)
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onForegroundExited() {
        info(TAG, "onForegroundExited()")
        isInForeground.set(false)
    }

}